doc/cephfs/purge-queue.rst

```diff
@@ -0,0 +1,83 @@
+============
+Purge Queue
+============
+
+MDS maintains a data structure known as **Purge Queue** which is responsible
+for managing and executing the sequential deletion of files.
```

Comment:
Are you sure the deletion of files in sequential? What does `mds_max_purge_files` specify then?

Reply:
Looking at that parameter which says:
```
# Maximum number of concurrent stray files to purge
- name: mds_max_purge_files
  type: uint
  level: advanced
  desc: maximum number of deleted files to purge in parallel
  default: 64
  services:
  - mds
  with_legacy: true
```
it does but when I take a look at the code, especially https://github.com/ceph/ceph/blob/5061b31102daaa52b56749252e81aaccb8a3ab83/src/mds/PurgeQueue.cc#L624-L627

It maintains a ops vector which is queued into a finisher thread. Now going as per the Finisher basics, it does complete the contexts _asynchronously_ but that's still a queue, isn't it? That's the reason i put the deletion of files is sequential.

Reply:
Right. But if you see `PurgeQueue::_commit_ops`, it called into objecter with a new (gather) context which essentially would lead to multiple objecter calls.

Reply:
s/sequential/parallel - done, PTAL